src/logic/player.cc

```diff
@@ -1547,6 +1547,15 @@ void Player::hide_or_reveal_field(const Coords& coords, HideOrRevealFieldMode mo
 		assert(!field.vision.is_revealed());
 		assert(field.vision == VisibleState::kUnexplored);
 		break;
+
+	case HideOrRevealFieldMode::kUnexplore:
+		hide_or_reveal_field(coords, HideOrRevealFieldMode::kUnreveal);
+		if (field.vision == VisibleState::kPreviouslySeen) {
+			field.vision = VisibleState::kUnexplored;
+		}
+		assert(!field.vision.is_revealed());
+		assert(field.vision == VisibleState::kUnexplored);
```

Comment:
These assertions do not hold if the fields are still seen by a bob. To reproduce just comment out the `ship:remove()` and you get
```
widelands: ../src/logic/player.cc:1557: void Widelands::Player::hide_or_reveal_field(const Widelands::Coords&, Widelands::HideOrRevealFieldMode): Assertion `field.vision == VisibleState::kUnexplored' failed.
```

Reply:
ok, the first one will hold the second one not. this was a wrong assumption from me. will take care of that tonight. same for the documentation.